### PR TITLE
Embedding support for offscreen-rendering

### PR DESCRIPTION
**Pull request by jdm**
The current embedding API assumes it should render directly to a native widget at all times. These changes expose an option to render to an offscreen surface when starting the engine, as well as a new WebrenderSurfman API for temporarily acquiring  the surface corresponding to the front buffer so it can be used for blitting purposes.

Demonstrated to work in https://github.com/jdm/servo-embedding-example.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

### CODE REVIEW

**Comment by jdm** [comment]
The first two commits are from #27717 and can be ignored.

**Comment by asajeffrey** [comment]
Ah, interesting. My first reaction was "didn't I already do this?" since it's needed for the gstreamer servosrc plugin, but that's because I built that using libservo not libsimpleservo.

You can r=me.

File: Cargo.lock [hunk 1]
```diff
@@ -7041,10 +7041,16 @@ dependencies = [
...
+name = "weezl"
+version = "0.1.3"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "3e2bb9fc8309084dd7cd651336673844c1d47f8ef6d2091ec160b27f5c4aa277"
+
 [[package]]
 name = "wgpu-core"
 version = "0.6.0"
-source = "git+https://github.com/gfx-rs/wgpu#e72724a6e393503c73f37e86aa9317a5c62e32b8"
+source = "git+https://github.com/gfx-rs/wgpu?rev=e72724a6e393503c73f37e86aa9317a5c62e32b8#e72724a6e393503c73f37e86aa9317a5c62e32b8"
```
Comment (atouchet) [inline Cargo.lock:7053]: Is it possible to update these to wgpu-core 0.6.5/wgpu-types 0.6.1 instead or is there an issue with using the current published versions in Servo?

Comment (jdm) [reply Cargo.lock:7053]: There are compilation errors using more recent git revisions. I have not attempted to investigate them at all.

**Pull request merged**